master/buildbot/www/hooks/github.py

```diff
@@ -230,20 +231,27 @@ def _get_commit_msg(self, repo, sha):
         :param repo: the repo full name, ``{owner}/{project}``.
             e.g. ``buildbot/buildbot``
         '''
+        if not self._token:
+            return 'No message field'
+
         headers = {
-            'User-Agent': 'Buildbot'
+            'Authorization': 'token ' + self._token,
+            'User-Agent': 'Buildbot',
         }
-        if self._token:
-            headers['Authorization'] = 'token ' + self._token
 
         url = '/repos/{}/commits/{}'.format(repo, sha)
         http = yield httpclientservice.HTTPClientService.getService(
             self.master, self.github_api_endpoint, headers=headers,
             debug=self.debug, verify=self.verify)
         res = yield http.get(url)
-        data = yield res.json()
-        msg = data.get('commit', {'message': 'No message field'})['message']
-        return msg
+        if 200 <= res.code < 300:
+            data = yield res.json()
+            return data['commit']['message']
+        else:
```

Comment:
I think this else is not needed, we can just fall through o the end of the function.

Reply:
Also, coverage results show that this error path is not covered by any tests. Can we add a test that covers it?

You may see "unhandled exception in logs" or similar errors after writing a test. To fix this, you'll need to add an expectation of failures in the logs like this: `self.assertEqual(len(self.flushLoggedErrors()), 1)`.

Reply:
Done, not sure why I ended up adding the `else:` but it's been adjusted.